tinggal styling discount price

adirasakhi opened the pull request.

andikarust commented:
<?php

namespace App\Http\Controllers;

use Carbon\Carbon;
use App\Models\User;
use App\Models\Product;
use App\Models\ProductClick;
use App\Models\Category;
use App\Models\SocialMedia;
use App\Models\Variant;
use Illuminate\Http\Request;
use Illuminate\Support\Facades\DB;

class KatalogController extends Controller
{
    public function index(Request $request)
    {
        $categoryId = $request->input('id');
        $minPrice = $request->input('min');
        $maxPrice = $request->input('max');
        $keywords = $request->input('keywords');

        $categories = Category::withCount('products')->get();

        $subquery = DB::table('variants')
            ->select('product_id', DB::raw('MIN(price) as min_price'))
            ->groupBy('product_id');

        $query = DB::table('product')
            ->joinSub($subquery, 'min_variants', function ($join) {
                $join->on('product.id', '=', 'min_variants.product_id');
            })
            ->join('variants', function ($join) {
                $join->on('product.id', '=', 'variants.product_id')
                    ->whereColumn('variants.price', '=', 'min_variants.min_price');
            })
            ->join('users', 'product.seller_id', '=', 'users.id')
            ->select('product.*', 'variants.image as min_variant_image', 'users.name as seller_name', 'min_variants.min_price as min_price');

        // Memuat relasi Seller untuk setiap produk
        $products = $query->paginate(12);

        return view('pages.Landing.shop', compact('products', 'categories', 'categoryId', 'minPrice', 'maxPrice'));
    }


    public function detail($id)
    {
        // Temukan produk berdasarkan ID
        $product = Product::with('variants', 'seller')->find($id);

        // Redirect jika produk tidak ditemukan
        if (!$product) {
            return redirect()->route('katalog.index')->with('error', 'Product not found.');
        }

        // Ambil semua kategori
        $categories = Category::all();

        // Ambil semua pengguna (mungkin Anda ingin filter ini lebih spesifik)
        $users = User::all();

        // Ambil semua media sosial
        $social_media = SocialMedia::all();

        // Ambil produk terkait berdasarkan kategori yang sama, kecuali produk saat ini
        $subquery = DB::table('variants')
            ->select('product_id', DB::raw('MIN(price) as min_price'))
            ->groupBy('product_id');

        $related_products = DB::table('product')
            ->joinSub($subquery, 'min_variants', function ($join) {
                $join->on('product.id', '=', 'min_variants.product_id');
            })
            ->join('variants', function ($join) {
                $join->on('product.id', '=', 'variants.product_id')
                    ->whereColumn('variants.price', '=', 'min_variants.min_price');
            })
            ->join('users', 'product.seller_id', '=', 'users.id')
            ->where('product.category_id', $product->category_id)
            ->where('product.id', '!=', $product->id)
            ->select('product.*', 'variants.image as min_variant_image', 'users.name as seller_name', 'min_variants.min_price as min_price')
            ->limit(3)
            ->get();

        // Mengidentifikasi perangkat menggunakan IP address atau metode lain
        $deviceId = request()->ip();

        // Cek jika sudah ada klik untuk produk ini hari ini dari perangkat ini
        $existingClick = ProductClick::where('product_id', $id)
            ->where('device_id', $deviceId)
            ->whereDate('clicked_at', Carbon::today())
            ->first();

        if ($existingClick) {
            // Jika sudah ada, tambahkan jumlah klik
            $existingClick->increment('click_count');
        } else {
            // Jika belum ada, buat entri baru dengan click_count = 1
            ProductClick::create([
                'product_id' => $id,
                'device_id' => $deviceId,
                'clicked_at' => Carbon::now(),
                'click_count' => 1,
            ]);
        }

        // Menghitung harga setelah diskon
        foreach ($product->variants as $variant) {
            $variant->discounted_price = $variant->getDiscountedPriceAttribute();
        }

        // Kembalikan view dengan data yang diperlukan
        return view('pages.Landing.Detail', [
            'product' => $product,
            'categories' => $categories,
            'users' => $users,
            'social_media' => $social_media,
            'related_products' => $related_products,
            'variants' => $product->variants,
        ]);
    }


    public function filter(Request $request)
    {
        $categoryId = $request->input('id');
        $keywords = $request->input('keywords');
        $minPrice = $request->input('min');
        $maxPrice = $request->input('max');
        $sort = $request->input('sort', 'asc');

        // Subquery to get the minimum price and corresponding variant id for each product
        $subquery = DB::table('variants')
            ->select('product_id', DB::raw('MIN(price) as min_price'))
            ->groupBy('product_id');

        $query = DB::table('product')
            ->joinSub($subquery, 'min_variants', function ($join) {
                $join->on('product.id', '=', 'min_variants.product_id');
            })
            ->join('variants', function ($join) {
                $join->on('product.id', '=', 'variants.product_id')
                    ->whereColumn('variants.price', '=', 'min_variants.min_price');
            })
            ->join('users', 'product.seller_id', '=', 'users.id')
            ->select('product.*', 'variants.image as min_variant_image', 'users.name as seller_name', 'min_variants.min_price as min_price');

        if (isset($categoryId) && ($categoryId != null)) {
            $query->where('product.category_id', $categoryId);
        }

        if (isset($minPrice) && ($minPrice != null)) {
            $query->where('min_variants.min_price', '>=', $minPrice);
        }

        if (isset($maxPrice) && ($maxPrice != null)) {
            $query->where('min_variants.min_price', '<=', $maxPrice);
        }

        if (isset($keywords) && ($keywords != null)) {
            $keywordArray = explode(' ', $keywords);
            foreach ($keywordArray as $keyword) {
                $query->where('product.name', 'like', '%' . $keyword . '%');
            }
        }

        if (!in_array($sort, ['asc', 'desc'])) {
            $sort = 'asc';
        }

        $query->orderBy('min_variants.min_price', $sort);

        $products = $query->paginate(12);
        $categories = Category::withCount('products')->get();

        return view('pages.Landing.shop', compact('products', 'categories', 'minPrice', 'maxPrice', 'sort'));
    }

    public function search(Request $request)
    {
        $keywords = $request->input('keywords');
        $categoryId = $request->input('id');
        $minPrice = $request->input('min');
        $maxPrice = $request->input('max');
        $sort = $request->input('sort', 'asc');

        // Subquery untuk mendapatkan harga minimum dan id varian yang sesuai untuk setiap produk
        $subquery = DB::table('variants')
            ->select('product_id', DB::raw('MIN(price) as min_price'))
            ->groupBy('product_id');

        $query = DB::table('product')
            ->joinSub($subquery, 'min_variants', function ($join) {
                $join->on('product.id', '=', 'min_variants.product_id');
            })
            ->join('variants', function ($join) {
                $join->on('product.id', '=', 'variants.product_id')
                    ->whereColumn('variants.price', '=', 'min_variants.min_price');
            })
            ->join('users', 'product.seller_id', '=', 'users.id')
            ->select('product.*', 'variants.image as min_variant_image', 'users.name as seller_name', 'min_variants.min_price as min_price');

        if (!is_null($categoryId)) {
            $query->where('product.category_id', $categoryId);
        }

        if (!is_null($minPrice)) {
            $query->where('min_variants.min_price', '>=', $minPrice);
        }

        if (!is_null($maxPrice)) {
            $query->where('min_variants.min_price', '<=', $maxPrice);
        }

        if (!empty($keywords)) {
            $keywordArray = explode(' ', $keywords);
            foreach ($keywordArray as $keyword) {
                $query->where('product.name', 'like', '%' . $keyword . '%');
            }
        }

        if (!in_array($sort, ['asc', 'desc'])) {
            $sort = 'asc';
        }

        $query->orderBy('min_variants.min_price', $sort);

        $products = $query->paginate(12);
        $categories = Category::withCount('products')->get();

        return view('pages.Landing.shop', compact('products', 'categories', 'minPrice', 'maxPrice', 'sort'));
    }


    public function getPopularProduct()
    {
        $user = User::all();
        $popularProduct = Product::select(
            'product.id',
            'product.name',
            'product.price',
            'product.image',
            'users.name as saller_name',
            DB::raw('COUNT(product_clicks.id) as click_count')
        )
            ->join('product_clicks', 'product.id', '=', 'product_clicks.product_id')
            ->join('users', 'product.seller_id', '=', 'users.id')

            ->whereDate('product_clicks.clicked_at', '>=', Carbon::now()->subDays(30))
            ->groupBy(
                'product.id',
                'product.name',
                'product.price',
                'product.image',
                'users.name'
            )
            ->orderByDesc('click_count')
            ->take(10)
            ->get();


        return view('pages.Landing.index', ['popularProduct' => $popularProduct]);
    }
}

andikarust commented:
@extends('layouts.landingPage')
@section('content')

    <!-- Single Product Start -->
    <div class="container-fluid py-5 mt-5">
        <div class="container py-5">
            <div class="row g-4 mb-5">
                <!-- Main Image and Thumbnails -->
                <div class="col-lg-4">
                    <div class="main-image-container border rounded mb-3">
                        <a href="#">
                            <img src="{{ asset('storage/' . $product->variants->sortBy('price')->first()->image) }}"
                                class="img-fluid rounded" alt="Image" id="expandedImg" style="object-fit: cover;">
                        </a>
                    </div>
                    <div class="scrollable-row">
                        @foreach ($variants as $variant)
                            <div class="col">
                                <img src="{{ asset('storage/' . $variant->image) }}" class="img-fluid rounded imgClick"
                                    alt="Image" style="object-fit: cover;" data-variant-id="{{ $variant->id }}"
                                    data-variant-name="{{ $variant->name }}"
                                    data-variant-price="{{ number_format($variant->price, 0, ',', '.') }}"
                                    data-variant-discounted-price="{{ number_format($variant->discounted_price, 0, ',', '.') }}"
                                    data-variant-image="{{ asset('storage/' . $variant->image) }}">
                            </div>
                        @endforeach
                    </div>
                </div>
                <!-- Product Details -->
                <div class="col-lg-5 scrollable-column">
                    <h4 class="fw-bold mb-1">{{ $product->name }}</h4>
                    <p class="mb-1">{{ $product->category->category }}</p>
                    <a href="#" data-bs-toggle="modal" data-bs-target="#myModalSeller">
                        <p class="mb-2"><i class="fas fa-store"></i> {{ $product->seller->name }}</p>
                    </a>
                    <div>
                        @if (
                            $product->variants->sortBy('price')->first()->price >
                                $product->variants->sortBy('price')->first()->discounted_price)
                            <p class="strike-through">Rp{{ number_format($product->variants->sortBy('price')->first()->price, 0, ',', '.') }}</p>
                            <h2 class="discounted-price" id="orderVariantPrice">
                                Rp{{ number_format($product->variants->sortBy('price')->first()->discounted_price, 0, ',', '.') }}
                            </h2>
                        @else
                            <h2 class="discounted-price" id="orderVariantPrice">
                                Rp{{ number_format($product->variants->sortBy('price')->first()->price, 0, ',', '.') }}
                            </h2>
                        @endif
                    </div>
                    <h6><strong>Pilih Varian</strong></h6>
                    <div class="variant-container">
                        @foreach ($variants as $variant)
                            <div class="variant-item" data-variant-id="{{ $variant->id }}"
                                data-variant-name="{{ $variant->name }}"
                                data-variant-price="{{ number_format($variant->price, 0, ',', '.') }}"
                                data-variant-discounted-price="{{ number_format($variant->discounted_price, 0, ',', '.') }}"
                                data-variant-image="{{ asset('storage/' . $variant->image) }}">
                                {{ $variant->name }}
                            </div>
                        @endforeach
                    </div>
                    <div>
                        <div class="nav nav-tabs mb-3 justify-content-center">
                            <h6 class="nav-link active border-white border-bottom-0">
                                Deskripsi
                            </h6>
                        </div>
                    </div>
                    <div class="tab-content mb-5">
                        <div class="tab-pane active" id="nav-about" role="tabpanel" aria-labelledby="nav-about-tab">
                            <p style="text-align: justify">{!! $product->description !!}</p>
                        </div>
                    </div>
                </div>
                <!-- Order Section -->
                <div class="col-lg-3 border rounded" style="height:300px">
                    <div class="nav nav-tabs my-3 justify-content-center">
                        <h6 class="nav-link">
                            Atur Pesanan
                        </h6>
                    </div>
                    <div class="row">
                        <div class="d-flex my-3">
                            <img src="{{ asset('storage/' . $product->variants->sortBy('price')->first()->image) }}"
                                class="img-fluid" alt="Image" id="orderVariantImage"
                                style="object-fit: cover; width:30px; border-radius:4px;">
                            <div class="variant-item bg-white border-0"><strong
                                    id="orderVariantName">{{ $product->variants->sortBy('price')->first()->name }}</strong>
                            </div>
                        </div>
                    </div>
                    <div class="input-group quantity mb-3" style="width: 100px;">
                        <div class="input-group-btn">
                            <button class="btn btn-sm btn-minus rounded-circle bg-light border">
                                <i class="fa fa-minus"></i>
                            </button>
                        </div>
                        <input type="text" id="productQuantity" class="form-control form-control-sm text-center border-0"
                            value="1">
                        <div class="input-group-btn">
                            <button class="btn btn-sm btn-plus rounded-circle bg-light border">
                                <i class="fa fa-plus"></i>
                            </button>
                        </div>
                    </div>
                    <div class="d-flex justify-content-between">
                        <p>Total Pesanan</p>
                        <h5>
                            <strong id="orderVariantTotal" data-variant-id="{{ $variant->id }}"
                                data-variant-name="{{ $variant->name }}" data-variant-price="{{ $variant->price }}"
                                data-variant-discounted-price="{{ $variant->discounted_price }}"
                                data-variant-image="{{ asset('storage/' . $variant->image) }}">
                                Rp{{ number_format($product->variants->sortBy('discounted_price')->first()->discounted_price, 0, ',', '.') }}
                            </strong>
                        </h5>
                    </div>

                    <div class="d-flex contact-seller justify-content-center">
                        <a href="#" id="contactSellerBtn" class="btn-custom">
                            <i class="fa fa-phone me-2"></i> Hubungi Penjual
                        </a>
                    </div>
                </div>
            </div>
            <h1 class="fw-bold mb-0">Produk Terkait</h1>
            <div class="vesitable">
                <div class="owl-carousel vegetable-carousel justify-content-center">
                    @foreach ($related_products as $related_product)
                        <div class="card h-100">
                            <img src="{{ asset('storage/' . $related_product->min_variant_image) }}" class="card-img-top"
                                style="height: 150px; object-fit: cover;" alt="Image">
                            <div class="card-body">
                                <a href="{{ route('katalog.detail', ['id' => $related_product->id]) }}">
                                    <h6 class="card-title">{{ $related_product->name }}</h6>
                                </a>
                                <h6><strong>Rp{{ number_format($related_product->min_price, 0, ',', '.') }}</strong></h6>
                                <p class="small"><i class="fas fa-store"></i> {{ $related_product->seller_name }}</p>
                            </div>
                        </div>
                    @endforeach
                </div>
            </div>
        </div>
    </div>
    <!-- Single Product End -->

    <!-- Modal Info Toko -->
    <div class="modal fade" id="myModalSeller" role="dialog" aria-labelledby="myModalLabel">
        <div class="modal-dialog" role="document">
            <div class="modal-content">
                <div class="modal-header">
                    <h4 class="modal-title" id="myModalLabel">Informasi Toko</h4>
                </div>
                <div class="modal-body">
                    <div class="row">
                        <img src="{{ asset('storage/' . $product->seller->photo) }}" alt="">
                        <p><strong>Toko :</strong> {{ $product->seller->name }}</p>
                    </div>
                    <p><strong>Lokasi :</strong> {{ $product->seller->address }}</p>
                    <p><strong>Nomor Telepon :</strong> {{ $product->seller->phone }}</p>
                    <p><strong>Tanggal Bergabung :</strong>
                        {{ \Carbon\Carbon::parse($product->seller->created_at)->locale('id_ID')->isoFormat('DD MMMM YYYY') }}
                    </p>
                    <hr>
                    <p><strong>Media Sosial</strong></p>
                    <div>
                        @if ($product->seller->social)
                            @if ($product->seller->social->facebook)
                                <a href="https://www.facebook.com/{{ $product->seller->social->facebook }}"
                                    style="display: inline-block; margin-right: 10px;">
                                    <span class="badge bg-primary text-white" style="padding: 10px; border-radius: 5px;">
                                        <i class="bi bi-facebook text-white"></i> Facebook
                                    </span>
                                </a>
                            @endif
                            @if ($product->seller->social->instagram)
                                <a href="https://www.instagram.com/{{ $product->seller->social->instagram }}"
                                    style="display: inline-block; margin-right: 10px;">
                                    <span class="badge bg-danger text-white" style="padding: 10px; border-radius: 5px;">
                                        <i class="bi bi-instagram text-white"></i> Instagram
                                    </span>
                                </a>
                            @endif
                            @if ($product->seller->social->tiktok)
                                <a href="https://www.tiktok.com/@{{ $product - > seller - > social - > tiktok }}"
                                    style="display: inline-block; margin-right: 10px;">
                                    <span class="badge bg-dark text-white" style="padding: 10px; border-radius: 5px;">
                                        <i class="bi bi-tiktok text-white"></i> TikTok
                                    </span>
                                </a>
                            @endif
                            @if ($product->seller->social->youtube)
                                <a href="https://www.youtube.com/{{ $product->seller->social->youtube }}"
                                    style="display: inline-block; margin-right: 10px;">
                                    <span class="badge bg-danger text-white" style="padding: 10px; border-radius: 5px;">
                                        <i class="bi bi-youtube text-white"></i> YouTube
                                    </span>
                                </a>
                            @endif
                        @endif
                    </div>
                </div>
                <div class="modal-footer">
                    <a class="btn btn-secondary" data-bs-dismiss="modal">Tutup</a>
                </div>
            </div>
        </div>
    </div>

    <script>
        document.addEventListener('DOMContentLoaded', function() {
            const imgClickElements = document.querySelectorAll('.imgClick');
            const variantItems = document.querySelectorAll('.variant-item');
            const expandedImg = document.getElementById('expandedImg');
            const orderVariantImage = document.getElementById('orderVariantImage');
            const orderVariantName = document.getElementById('orderVariantName');
            const orderVariantPrice = document.getElementById('orderVariantPrice');
            const orderVariantTotal = document.getElementById('orderVariantTotal');
            const productQuantity = document.getElementById('productQuantity');
            const btnPlus = document.querySelector('.btn-plus');
            const btnMinus = document.querySelector('.btn-minus');
            const contactSellerBtn = document.getElementById('contactSellerBtn');

            imgClickElements.forEach(imgClick => {
                imgClick.addEventListener('click', function() {
                    updateVariantDetails(this);
                });
            });

            variantItems.forEach(variantItem => {
                variantItem.addEventListener('click', function() {
                    updateVariantDetails(this);
                });
            });

            btnPlus.addEventListener('click', function() {
                productQuantity.value = parseInt(productQuantity.value); // Increment quantity by 1
                updateTotalPrice();
            });

            btnMinus.addEventListener('click', function() {
                if (parseInt(productQuantity.value) > 0) {
                    productQuantity.value = parseInt(productQuantity
                    .value); // Decrement quantity by 1, but not below 1
                    updateTotalPrice();
                }
            });

            function updateVariantDetails(element) {
                const variantId = element.dataset.variantId;
                const variantName = element.dataset.variantName;
                const variantPrice = parseFloat(element.dataset.variantPrice.replaceAll('.', ''));
                const variantDiscountedPrice = parseFloat(element.dataset.variantDiscountedPrice.replaceAll('.',
                    ''));
                const variantImage = element.dataset.variantImage;
                expandedImg.src = variantImage;
                orderVariantImage.src = variantImage;
                orderVariantName.textContent = variantName;

                if (variantPrice > variantDiscountedPrice) {
                    orderVariantPrice.innerHTML =
                        `Rp${new Intl.NumberFormat('id-ID').format(variantPrice)}<br>Rp${new Intl.NumberFormat('id-ID').format(variantDiscountedPrice)}`;
                } else {
                    orderVariantPrice.textContent = `Rp${new Intl.NumberFormat('id-ID').format(variantPrice)}`;
                }

                updateTotalPrice();
            }

            function updateTotalPrice() {
                let quantity = parseInt(productQuantity.value);
                let priceElements = orderVariantPrice.innerHTML.split('<br>');
                let discountedPrice;

                if (priceElements.length > 1) {
                    discountedPrice = parseFloat(priceElements[1].replace(/[^\d]/g, ''));
                } else {
                    discountedPrice = parseFloat(priceElements[0].replace(/[^\d]/g, ''));
                }

                let total = discountedPrice * quantity;
                orderVariantTotal.textContent = `Rp${new Intl.NumberFormat('id-ID').format(total)}`;
            }

            // Initial update on page load
            updateTotalPrice();

            contactSellerBtn.addEventListener('click', function(event) {
                event.preventDefault();
                var quantity = productQuantity.value;
                var productName = "{{ $product->name }}";
                var variantName = orderVariantName.textContent;
                var priceElements = orderVariantPrice.innerHTML.split('<br>');
                var originalPrice = priceElements.length > 1 ? parseFloat(priceElements[0].replace(/[^\d]/g,
                    '')) : null;
                var variantPrice = parseFloat(priceElements[priceElements.length - 1].replace(/[^\d]/g,
                ''));
                var totalPrice = variantPrice * quantity;
                var sellerPhone = "{{ $product->seller->phone }}";

                var message = `Halo, saya tertarik dengan produk Anda.
    Nama produk: ${productName}
    Varian: ${variantName}
    Harga produk: ${originalPrice ? `Rp${originalPrice.toLocaleString('id-ID')} (diskon menjadi: Rp${variantPrice.toLocaleString('id-ID')})` : `Rp${variantPrice.toLocaleString('id-ID')}`}
    Jumlah produk yang dibeli: ${quantity}
    Total harga produk: Rp${totalPrice.toLocaleString('id-ID')}
    Link produk: ${location.href}`;

                var whatsappUrl = `https://wa.me/${sellerPhone}?text=${encodeURIComponent(message)}`;
                window.open(whatsappUrl, '_blank');
            });
        });
    </script>

@endsection

andikarust commented:
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <title>Fruitables - Vegetable Website Template</title>
    <meta content="width=device-width, initial-scale=1.0" name="viewport">
    <meta content="" name="keywords">
    <meta content="" name="description">

    <!-- Google Web Fonts -->
    <link rel="preconnect" href="https://fonts.googleapis.com">
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
    <link
        href="https://fonts.googleapis.com/css2?family=Open+Sans:wght@400;600&family=Raleway:wght@600;800&display=swap"
        rel="stylesheet">

    <!-- Icon Font Stylesheet -->
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.15.4/css/all.css" />
    <link href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.4.1/font/bootstrap-icons.css" rel="stylesheet">

    <!-- Libraries Stylesheet -->
    <link href="{{ asset('LandingPage/lib/lightbox/css/lightbox.min.css') }}" rel="stylesheet">
    <link href="{{ asset('LandingPage/lib/owlcarousel/assets/owl.carousel.min.css') }}" rel="stylesheet">


    <!-- Customized Bootstrap Stylesheet -->
    <link href="{{ asset('LandingPage/css/bootstrap.min.css') }}" rel="stylesheet">

    <!-- Template Stylesheet -->
    <link href="{{ asset('LandingPage/css/style.css') }}" rel="stylesheet">

    <style>
        .btn-custom {
            background-color: #404040;
            color: white;
            border: 1px solid #6c757d;
            border-radius: 50px;
            padding: 0.5rem 1rem;
            margin-bottom: 1rem;
            display: inline-flex;
            align-items: center;
            text-decoration: none;
        }

        .btn-custom:hover {
            background-color: black;
            color: white;
        }

        .btn-custom .fa-phone {
            color: white;
            margin-right: 0.5rem;
        }

        .fixed-top {
            width: 100%;
            background-color: white;
            z-index: 1030;
            /* Ensure it stays above other content */
        }


        @media (min-width: 853px) {
            #filterModalToggle {
                display: none;
            }
        }

        @media (max-width: 853px) {
            #filterContainer {
                display: none;
            }

            .user-icon {
                text-decoration: none !important;
                color: black !important;
            }

            .user-icon i {
                display: none;
            }

            .user-link {
                font-weight: bold;
                color: black;
            }
        }

        .modal-dialog-scrollable .modal-body {
            max-height: 54vh;
            overflow-y: auto;
        }

        .additional-category {
            margin-top: 0;
        }


        .custom-checkbox {
            border-radius: 3px;
            width: 18px;
            height: 18px;
            appearance: none;
            -webkit-appearance: none;
            background-color: #fff;
            border: 1px solid #ced4da;
            outline: none;
            cursor: pointer;
            position: relative;
            display: inline-block;
            vertical-align: middle;
        }

        .custom-checkbox:checked::before {
            content: "\2714";
            position: absolute;
            top: 50%;
            left: 50%;
            transform: translate(-50%, -50%);
            font-size: 14px;
            color: black;
        }

        .custom-checkbox:hover {
            border-color: black;
        }

        .scrollable-row {
            display: flex;
            overflow-x: auto;
            padding: 10px;
            white-space: nowrap;
        }

        .main-image-container img {
            height: 300px;
            width: 100%;
            object-fit: cover;
            border-radius: 8px;
        }

        .scrollable-row {
            display: flex;
            overflow-x: auto;
            padding: 10px;
            white-space: nowrap;
            gap: 10px;
        }

        .scrollable-row .col {
            flex: 0 0 auto;
            border: 1px solid #ddd;
            border-radius: 8px;
        }

        .scrollable-row .col img {
            height: 100px;
            width: 100px;
            object-fit: cover;
            border-radius: 8px;
        }

        .variant-container {
            display: flex;
            flex-wrap: wrap;
            gap: 10px;
            padding: 10px 0;
        }

        .variant-item {
            flex: 0 0 auto;
            white-space: nowrap;
            padding: 5px 10px;
            border: 1px solid #ddd;
            border-radius: 5px;
            text-align: center;
            background-color: #f8f8f8;
        }

        .variant-item:hover {
            background-color: #e2e2e2;
            cursor: pointer;
        }

        .main-image-container img {
            width: 100%;
            height: 360px;
        }

        .nav-tabs .nav-link {
            text-align: center;
            position: relative;
            display: inline-block;
            padding-bottom: 10px;
        }

        .nav-tabs .nav-link::after {
            content: '';
            display: block;
            width: 100%;
            height: 2px;
            background-color: black;
            position: absolute;
            bottom: 0;
            left: 0;
        }

        .scrollable-column {
            max-height: 500px;
            /* Adjust this value as needed */
            overflow-y: auto;
        }

        h6,
        h2#orderVariantPrice {
            margin-bottom: 0;
            /* Menghapus margin bawah dari elemen h6 */
            margin-top: 0;
            /* Menghapus margin atas dari elemen h2 */
        }

        .strike-through {
            text-decoration: line-through;
            opacity: 0.6;
            font-weight: lighter;
        }
    </style>
</head>

<body>

    <!-- Spinner Start -->
    <div id="spinner"
        class="show w-100 vh-100 bg-white position-fixed translate-middle top-50 start-50  d-flex align-items-center justify-content-center">
        <div class="spinner-grow text-primary" role="status"></div>
    </div>
    <!-- Spinner End -->


    <!-- Modal Search Start -->
    <div class="modal fade" id="searchModal" tabindex="-1" aria-labelledby="exampleModalLabel" aria-hidden="true">
        <div class="modal-dialog modal-fullscreen">
            <div class="modal-content rounded-0">
                <div class="modal-header">
                    <h5 class="modal-title" id="exampleModalLabel">Search by keyword</h5>
                    <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
                </div>
                <div class="modal-body d-flex align-items-center">
                    <div class="input-group w-75 mx-auto d-flex">
                        <input type="search" class="form-control p-3" placeholder="keywords"
                            aria-describedby="search-icon-1">
                        <span id="search-icon-1" class="input-group-text p-3"><i class="fa fa-search"></i></span>
                    </div>
                </div>
            </div>
        </div>
    </div>
    <!-- Modal Search End -->
    <!-- Navbar start -->
    @include('partials.Landing.navbar')
    <!-- Navbar End -->
    <!-- Content Start -->
    <div class="container">
        @yield('content')
    </div>
    <!-- Content End -->

    <!-- Footer Start -->
    @include('partials.Landing.footer')
    <!-- Footer End -->




    <!-- Back to Top -->
    <a href="#" class="btn btn-primary border-3 border-primary rounded-circle back-to-top"><i
            class="fa fa-arrow-up" style="color: white"></i></a>


    <!-- JavaScript Libraries -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.4/jquery.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0/dist/js/bootstrap.bundle.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/OwlCarousel2/2.3.4/owl.carousel.min.js"></script>
    <script src="{{ asset('LandingPage/lib/easing/easing.min.js') }}"></script>
    <script src="{{ asset('LandingPage/lib/waypoints/waypoints.min.js') }}"></script>
    <script src="{{ asset('LandingPage/lib/lightbox/js/lightbox.min.js') }}"></script>
    {{-- <script src="https://code.jquery.com/jquery-3.5.1.slim.min.js"></script> --}}
    <script src="https://cdn.jsdelivr.net/npm/@popperjs/core@2.5.4/dist/umd/popper.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.5.2/js/bootstrap.min.js"></script>


    <!-- Template Javascript -->
    <script src="{{ asset('LandingPage/js/main.js') }}"></script>

    <script>
        $(document).ready(function() {

            $(".owl-carousel").owlCarousel({
                items: 3,
                loop: true,
                margin: 10,
                nav: true,
                dots: true,
                autoplay: true,
                autoplayTimeout: 3000,
                autoplayHoverPause: true
            });
        });
    </script>
</body>

</html>

adirasakhi commented:
@extends('layouts.landingPage')
@section('content')

    <!-- Single Product Start -->
    <div class="container-fluid py-5 mt-5">
        <div class="container py-5">
            <div class="row g-4 mb-5">
                <!-- Main Image and Thumbnails -->
                <div class="col-lg-4">
                    <div class="main-image-container border rounded mb-3">
                        <a href="#">
                            <img src="{{ asset('storage/' . $product->variants->sortBy('price')->first()->image) }}"
                                class="img-fluid rounded" alt="Image" id="expandedImg" style="object-fit: cover;">
                        </a>
                    </div>
                    <div class="scrollable-row">
                        @foreach ($variants as $variant)
                            <div class="col">
                                <img src="{{ asset('storage/' . $variant->image) }}" class="img-fluid rounded imgClick"
                                    alt="Image" style="object-fit: cover;" data-variant-id="{{ $variant->id }}"
                                    data-variant-name="{{ $variant->name }}"
                                    data-variant-price="{{ number_format($variant->price, 0, ',', '.') }}"
                                    data-variant-discounted-price="{{ number_format($variant->discounted_price, 0, ',', '.') }}"
                                    data-variant-image="{{ asset('storage/' . $variant->image) }}">
                            </div>
                        @endforeach
                    </div>
                </div>
                <!-- Product Details -->
                <div class="col-lg-5 scrollable-column">
                    <h4 class="fw-bold mb-1">{{ $product->name }}</h4>
                    <p class="mb-1">{{ $product->category->category }}</p>
                    <a href="#" data-bs-toggle="modal" data-bs-target="#myModalSeller">
                        <p class="mb-2"><i class="fas fa-store"></i> {{ $product->seller->name }}</p>
                    </a>
                    <div>
                        @if (
                            $product->variants->sortBy('price')->first()->price >
                                $product->variants->sortBy('price')->first()->discounted_price)
                            <p class="strike-through">Rp{{ number_format($product->variants->sortBy('price')->first()->price, 0, ',', '.') }}</p>
                            <h2 class="discounted-price" id="orderVariantPrice">
                                Rp{{ number_format($product->variants->sortBy('price')->first()->discounted_price, 0, ',', '.') }}
                            </h2>
                        @else
                            <h2 class="discounted-price" id="orderVariantPrice">
                                Rp{{ number_format($product->variants->sortBy('price')->first()->price, 0, ',', '.') }}
                            </h2>
                        @endif
                    </div>
                    <h6><strong>Pilih Varian</strong></h6>
                    <div class="variant-container">
                        @foreach ($variants as $variant)
                            <div class="variant-item" data-variant-id="{{ $variant->id }}"
                                data-variant-name="{{ $variant->name }}"
                                data-variant-price="{{ number_format($variant->price, 0, ',', '.') }}"
                                data-variant-discounted-price="{{ number_format($variant->discounted_price, 0, ',', '.') }}"
                                data-variant-image="{{ asset('storage/' . $variant->image) }}">
                                {{ $variant->name }}
                            </div>
                        @endforeach
                    </div>
                    <div>
                        <div class="nav nav-tabs mb-3 justify-content-center">
                            <h6 class="nav-link active border-white border-bottom-0">
                                Deskripsi
                            </h6>
                        </div>
                    </div>
                    <div class="tab-content mb-5">
                        <div class="tab-pane active" id="nav-about" role="tabpanel" aria-labelledby="nav-about-tab">
                            <p style="text-align: justify">{!! $product->description !!}</p>
                        </div>
                    </div>
                </div>
                <!-- Order Section -->
                <div class="col-lg-3 border rounded" style="height:300px">
                    <div class="nav nav-tabs my-3 justify-content-center">
                        <h6 class="nav-link">
                            Atur Pesanan
                        </h6>
                    </div>
                    <div class="row">
                        <div class="d-flex my-3">
                            <img src="{{ asset('storage/' . $product->variants->sortBy('price')->first()->image) }}"
                                class="img-fluid" alt="Image" id="orderVariantImage"
                                style="object-fit: cover; width:30px; border-radius:4px;">
                            <div class="variant-item bg-white border-0"><strong
                                    id="orderVariantName">{{ $product->variants->sortBy('price')->first()->name }}</strong>
                            </div>
                        </div>
                    </div>
                    <div class="input-group quantity mb-3" style="width: 100px;">
                        <div class="input-group-btn">
                            <button class="btn btn-sm btn-minus rounded-circle bg-light border">
                                <i class="fa fa-minus"></i>
                            </button>
                        </div>
                        <input type="text" id="productQuantity" class="form-control form-control-sm text-center border-0"
                            value="1">
                        <div class="input-group-btn">
                            <button class="btn btn-sm btn-plus rounded-circle bg-light border">
                                <i class="fa fa-plus"></i>
                            </button>
                        </div>
                    </div>
                    <div class="d-flex justify-content-between">
                        <p>Total Pesanan</p>
                        <h5>
                            <strong id="orderVariantTotal" data-variant-id="{{ $variant->id }}"
                                data-variant-name="{{ $variant->name }}" data-variant-price="{{ $variant->price }}"
                                data-variant-discounted-price="{{ $variant->discounted_price }}"
                                data-variant-image="{{ asset('storage/' . $variant->image) }}">
                                Rp{{ number_format($product->variants->sortBy('discounted_price')->first()->discounted_price, 0, ',', '.') }}
                            </strong>
                        </h5>
                    </div>

                    <div class="d-flex contact-seller justify-content-center">
                        <a href="#" id="contactSellerBtn" class="btn-custom">
                            <i class="fa fa-phone me-2"></i> Hubungi Penjual
                        </a>
                    </div>
                </div>
            </div>
            <h1 class="fw-bold mb-0">Produk Terkait</h1>
            <div class="vesitable">
                <div class="owl-carousel vegetable-carousel justify-content-center">
                    @foreach ($related_products as $related_product)
                        <div class="card h-100">
                            <img src="{{ asset('storage/' . $related_product->min_variant_image) }}" class="card-img-top"
                                style="height: 150px; object-fit: cover;" alt="Image">
                            <div class="card-body">
                                <a href="{{ route('katalog.detail', ['id' => $related_product->id]) }}">
                                    <h6 class="card-title">{{ $related_product->name }}</h6>
                                </a>
                                <h6><strong>Rp{{ number_format($related_product->min_price, 0, ',', '.') }}</strong></h6>
                                <p class="small"><i class="fas fa-store"></i> {{ $related_product->seller_name }}</p>
                            </div>
                        </div>
                    @endforeach
                </div>
            </div>
        </div>
    </div>
    <!-- Single Product End -->

    <!-- Modal Info Toko -->
    <div class="modal fade" id="myModalSeller" role="dialog" aria-labelledby="myModalLabel">
        <div class="modal-dialog" role="document">
            <div class="modal-content">
                <div class="modal-header">
                    <h4 class="modal-title" id="myModalLabel">Informasi Toko</h4>
                </div>
                <div class="modal-body">
                    <div class="row">
                        <img src="{{ asset('storage/' . $product->seller->photo) }}" alt="">
                        <p><strong>Toko :</strong> {{ $product->seller->name }}</p>
                    </div>
                    <p><strong>Lokasi :</strong> {{ $product->seller->address }}</p>
                    <p><strong>Nomor Telepon :</strong> {{ $product->seller->phone }}</p>
                    <p><strong>Tanggal Bergabung :</strong>
                        {{ \Carbon\Carbon::parse($product->seller->created_at)->locale('id_ID')->isoFormat('DD MMMM YYYY') }}
                    </p>
                    <hr>
                    <p><strong>Media Sosial</strong></p>
                    <div>
                        @if ($product->seller->socialMedia)
                            @if ($product->seller->socialMedia->facebook)
                                <a href="https://www.facebook.com/{{ $product->seller->socialMedia->facebook }}"
                                    style="display: inline-block; margin-right: 10px;">
                                    <span class="badge bg-primary text-white" style="padding: 10px; border-radius: 5px;">
                                        <i class="bi bi-facebook text-white"></i> Facebook
                                    </span>
                                </a>
                            @endif
                            @if ($product->seller->socialMedia->instagram)
                                <a href="https://www.instagram.com/{{ $product->seller->socialMedia->instagram }}"
                                    style="display: inline-block; margin-right: 10px;">
                                    <span class="badge bg-danger text-white" style="padding: 10px; border-radius: 5px;">
                                        <i class="bi bi-instagram text-white"></i> Instagram
                                    </span>
                                </a>
                            @endif
                            @if ($product->seller->socialMedia->tiktok)
                                <a href="https://www.tiktok.com/@{{ $product->seller->socialMedia->tiktok }}"
                                    style="display: inline-block; margin-right: 10px;">
                                    <span class="badge bg-dark text-white" style="padding: 10px; border-radius: 5px;">
                                        <i class="bi bi-tiktok text-white"></i> TikTok
                                    </span>
                                </a>
                            @endif
                            @if ($product->seller->socialMedia->youtube)
                                <a href="https://www.youtube.com/{{ $product->seller->socialMedia->youtube }}"
                                    style="display: inline-block; margin-right: 10px;">
                                    <span class="badge bg-danger text-white" style="padding: 10px; border-radius: 5px;">
                                        <i class="bi bi-youtube text-white"></i> YouTube
                                    </span>
                                </a>
                            @endif
                        @endif
                    </div>
                </div>
                <div class="modal-footer">
                    <a class="btn btn-secondary" data-bs-dismiss="modal">Tutup</a>
                </div>
            </div>
        </div>
    </div>

    <script>
        document.addEventListener('DOMContentLoaded', function() {
            const imgClickElements = document.querySelectorAll('.imgClick');
            const variantItems = document.querySelectorAll('.variant-item');
            const expandedImg = document.getElementById('expandedImg');
            const orderVariantImage = document.getElementById('orderVariantImage');
            const orderVariantName = document.getElementById('orderVariantName');
            const orderVariantPrice = document.getElementById('orderVariantPrice');
            const orderVariantTotal = document.getElementById('orderVariantTotal');
            const productQuantity = document.getElementById('productQuantity');
            const btnPlus = document.querySelector('.btn-plus');
            const btnMinus = document.querySelector('.btn-minus');
            const contactSellerBtn = document.getElementById('contactSellerBtn');

            imgClickElements.forEach(imgClick => {
                imgClick.addEventListener('click', function() {
                    updateVariantDetails(this);
                });
            });

            variantItems.forEach(variantItem => {
                variantItem.addEventListener('click', function() {
                    updateVariantDetails(this);
                });
            });

            btnPlus.addEventListener('click', function() {
                productQuantity.value = parseInt(productQuantity.value); // Increment quantity by 1
                updateTotalPrice();
            });

            btnMinus.addEventListener('click', function() {
                if (parseInt(productQuantity.value) > 0) {
                    productQuantity.value = parseInt(productQuantity
                    .value); // Decrement quantity by 1, but not below 1
                    updateTotalPrice();
                }
            });

            function updateVariantDetails(element) {
                const variantId = element.dataset.variantId;
                const variantName = element.dataset.variantName;
                const variantPrice = parseFloat(element.dataset.variantPrice.replaceAll('.', ''));
                const variantDiscountedPrice = parseFloat(element.dataset.variantDiscountedPrice.replaceAll('.',
                    ''));
                const variantImage = element.dataset.variantImage;
                expandedImg.src = variantImage;
                orderVariantImage.src = variantImage;
                orderVariantName.textContent = variantName;

                if (variantPrice > variantDiscountedPrice) {
                    orderVariantPrice.innerHTML =
                        `Rp${new Intl.NumberFormat('id-ID').format(variantPrice)}<br>Rp${new Intl.NumberFormat('id-ID').format(variantDiscountedPrice)}`;
                } else {
                    orderVariantPrice.textContent = `Rp${new Intl.NumberFormat('id-ID').format(variantPrice)}`;
                }

                updateTotalPrice();
            }

            function updateTotalPrice() {
                let quantity = parseInt(productQuantity.value);
                let priceElements = orderVariantPrice.innerHTML.split('<br>');
                let discountedPrice;

                if (priceElements.length > 1) {
                    discountedPrice = parseFloat(priceElements[1].replace(/[^\d]/g, ''));
                } else {
                    discountedPrice = parseFloat(priceElements[0].replace(/[^\d]/g, ''));
                }

                let total = discountedPrice * quantity;
                orderVariantTotal.textContent = `Rp${new Intl.NumberFormat('id-ID').format(total)}`;
            }

            // Initial update on page load
            updateTotalPrice();

            contactSellerBtn.addEventListener('click', function(event) {
                event.preventDefault();
                var quantity = productQuantity.value;
                var productName = "{{ $product->name }}";
                var variantName = orderVariantName.textContent;
                var priceElements = orderVariantPrice.innerHTML.split('<br>');
                var originalPrice = priceElements.length > 1 ? parseFloat(priceElements[0].replace(/[^\d]/g,
                    '')) : null;
                var variantPrice = parseFloat(priceElements[priceElements.length - 1].replace(/[^\d]/g,
                ''));
                var totalPrice = variantPrice * quantity;
                var sellerPhone = "{{ $product->seller->socialMedia->whatsapp }}";

                var message = `Halo, saya tertarik dengan produk Anda.
    Nama produk: ${productName}
    Varian: ${variantName}
    Harga produk: ${originalPrice ? `Rp${originalPrice.toLocaleString('id-ID')} (diskon menjadi: Rp${variantPrice.toLocaleString('id-ID')})` : `Rp${variantPrice.toLocaleString('id-ID')}`}
    Jumlah produk yang dibeli: ${quantity}
    Total harga produk: Rp${totalPrice.toLocaleString('id-ID')}
    Link produk: ${location.href}`;

                var whatsappUrl = `https://wa.me/${sellerPhone}?text=${encodeURIComponent(message)}`;
                window.open(whatsappUrl, '_blank');
            });
        });
    </script>

@endsection
